fv3core/stencils/divergence_damping.py

```diff
@@ -149,90 +149,138 @@ class DivergenceDamping:
     """
 
     def __init__(
-        self, namelist: SimpleNamespace, nord_col: FloatFieldK, d2_bg: FloatFieldK
+        self,
+        grid_indexing: GridIndexing,
+        agrid1,
+        agrid2,
+        bgrid1,
+        bgrid2,
+        dxa,
+        dya,
+        edge_n,
+        edge_s,
+        edge_e,
+        edge_w,
+        nested: bool,
+        stretched_grid: bool,
+        da_min,
+        da_min_c,
+        divg_u,
+        divg_v,
+        rarea_c,
+        sin_sg1,
+        sin_sg2,
+        sin_sg3,
+        sin_sg4,
+        cosa_u,
+        cosa_v,
+        sina_u,
+        sina_v,
+        dxc,
+        dyc,
+        dddmp,
+        d4_bg,
+        nord,
+        grid_type,
+        nord_col: FloatFieldK,
+        d2_bg: FloatFieldK,
     ):
-        self.grid = spec.grid
-        assert not self.grid.nested, "nested not implemented"
-        assert (
-            namelist.grid_type < 3
-        ), "Not implemented, grid_type>=3, specifically smag_corner"
-        self._dddmp = namelist.dddmp
-        self._d4_bg = namelist.d4_bg
-        self._grid_type = namelist.grid_type
+        self._idx = grid_indexing
+        assert not nested, "nested not implemented"
+        assert grid_type < 3, "Not implemented, grid_type>=3, specifically smag_corner"
+        # TODO: make dddmp a compile-time external, instead of runtime scalar
+        self._dddmp = dddmp
+        # TODO: make da_min_c a compile-time external, instead of runtime scalar
+        self._da_min_c = da_min_c
+        self._grid_type = grid_type
         self._nord_column = nord_col
         self._d2_bg_column = d2_bg
+        self._divg_u = divg_u
+        self._divg_v = divg_v
+        self._rarea_c = rarea_c
+        self._sin_sg1 = sin_sg1
+        self._sin_sg2 = sin_sg2
+        self._sin_sg3 = sin_sg3
+        self._sin_sg4 = sin_sg4
+        self._cosa_u = cosa_u
+        self._cosa_v = cosa_v
+        self._sina_u = sina_u
+        self._sina_v = sina_v
+        self._dxc = dxc
+        self._dyc = dyc
         self._nonzero_nord_k = 0
-        self._nonzero_nord = int(namelist.nord)
+        self._nonzero_nord = int(nord)
         for k in range(len(self._nord_column)):
             if self._nord_column[k] > 0:
                 self._nonzero_nord_k = k
                 self._nonzero_nord = int(self._nord_column[k])
                 break
-        if self.grid.stretched_grid:
-            self._dd8 = self.grid.da_min * self._d4_bg ** (self._nonzero_nord + 1)
+        if stretched_grid:
+            self._dd8 = da_min * d4_bg ** (self._nonzero_nord + 1)
```

Comment:
I'd suggest that we use the translate classes to keep track of this, since they grab these from grid/namelist when calling the inits and can't be wrong.